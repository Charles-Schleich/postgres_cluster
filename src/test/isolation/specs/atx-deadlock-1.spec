setup
{
  create table a();
  create table b();
}

teardown
{
  drop table a;
  drop table b;
}

session "s1"
setup         { begin; SET deadlock_timeout = '10s'; }
step "s1la"   { lock table a in access exclusive mode; }
step "s1atxb" { begin autonomous; }
step "s1lb"   { lock table b in access exclusive mode; }
step "s1atxc" { commit; }
step "s1c"    { commit; }

session "s2"
setup       { begin; SET deadlock_timeout = '10s'; }
step "s2lb" { lock table b in access exclusive mode; }
step "s2la" { lock table a in access exclusive mode; }
step "s2c"  { commit; }

permutation "s1la" "s2lb" "s1atxb" "s2la" "s1lb" "s1atxc" "s1c" "s2c"
