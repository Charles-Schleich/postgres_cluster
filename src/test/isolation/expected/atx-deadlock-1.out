Parsed test spec with 2 sessions

starting permutation: s1la s2lb s1atxb s2la s1lb s1atxc s1c s2c
step s1la: lock table a in access exclusive mode;
step s2lb: lock table b in access exclusive mode;
step s1atxb: begin autonomous;
step s2la: lock table a in access exclusive mode; <waiting ...>
step s1lb: lock table b in access exclusive mode; <waiting ...>
step s1lb: <... completed>
step s2la: <... completed>
error in steps s1lb s2la: ERROR:  deadlock detected
step s1atxc: commit;
step s1c: commit;
step s2c: commit;
